### PR TITLE
chore: upgrade to v1.14.0

### PR DESCRIPTION
**Pull request by chungquantin**
Status: ✅ Compiled and built successfully

- ISSUE: https://github.com/r0gue-io/base-parachain/issues/39
- Release: https://github.com/paritytech/polkadot-sdk/releases/tag/polkadot-v1.14.0

- Compare [Parachain Template v1.13.0](https://github.com/paritytech/polkadot-sdk/commits/polkadot-v1.13.0/templates/parachain) vs [Parachain Template v1.14.0](https://github.com/paritytech/polkadot-sdk/commits/polkadot-v1.14.0/templates/parachain)

## Direct changes
- Introduce the construct runtime V2 (https://github.com/r0gue-io/base-parachain/issues/37)
- [[#4730](https://github.com/paritytech/polkadot-sdk/pull/4730)]: rpc upgrade jsonrpsee to v0.23.1
- [[#4769](https://github.com/paritytech/polkadot-sdk/pull/4769)]: Use real rust type for pallet alias in `runtime` macro
- Use aggregated types for `RuntimeFreezeReason`: https://github.com/paritytech/polkadot-sdk/commit/988103d7578ad515b13c69578da1237b28fa9f36

### CODE REVIEW

**Review by al3mart — COMMENTED** [review]
Could be sensible to ship the template wiith `MigrateToLatestXcmVersion` baked into the runtime.

I'd like to read others opinions, though.

**Review by Daanvdplas — CHANGES_REQUESTED** [review]
I see dependencies that are bumped when looking at the Cargo.toml file of the SDK on release v1.14 (e.g. codec = 3.6.12), compared to the ones here. Please compare the ones above line 40 (`#Substrate`).

service.rs:
```
+use prometheus_endpoint::Registry;
-use substrate_prometheus_endpoint::Registry;
```

runtime/src/config/mod.rs:
```
message_queue
-    type HeapSize = sp_core::ConstU32<{ 64 * 1024 }>;
+    type HeapSize = sp_core::ConstU32<{ 103 * 1024 }>;
```
xcm.rs
```
-    type FeeManager = XcmFeeManagerFromComponents<
-        WaivedLocations,
-        // Delivery fees are sent to the treasury account.
-        // These funds are not accessible without a module controlling such an account.
-        // [pallet_treasury](https://github.com/paritytech/polkadot-sdk/tree/master/substrate/frame/treasury)
-        // could be suitable option, configured using the same `TREASURY_PALLET_ID` used above as the pallet config's `PalletId`.
-        XcmFeeToAccount<Self::AssetTransactor, AccountId, TreasuryAccount>,
-    >;
+    type FeeManager = ();

-    type XcmTeleportFilter = Nothing;
+    type XcmTeleportFilter = Everything;
```

> Could be sensible to ship the template wiith MigrateToLatestXcmVersion baked into the runtime.

I see no harm in putting it in there.

**Review by al3mart — COMMENTED** [review]
Can't find anything more than what Daan has spotted.
Left a pair of super small nits.

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -117,13 +114,17 @@ pub type SignedExtra = (
 pub type UncheckedExtrinsic =
     generic::UncheckedExtrinsic<Address, RuntimeCall, Signature, SignedExtra>;
 
+/// Migrations to apply on runtime upgrade.
+pub type Migrations = (pallet_xcm::migration::MigrateToLatestXcmVersion<Runtime>);
```
[the remark this answers is not in the document]
Comment (chungquantin) [reply runtime/src/lib.rs]: Thank you @al3mart, I fixed this.

**Review by al3mart — APPROVED** [review]

**Pull request merged**